src/screenshot.cpp

```diff
@@ -856,3 +864,83 @@ bool MakeScreenshot(ScreenshotType t, const char *name)
 
 	return ret;
 }
+
+static byte _owner_colours[OWNER_END + 1];
+
+/**
+ * Return the colour a tile would be displayed with in the small map in mode "Owner".
+ *
+ * @param tile The tile of which we would like to get the colour.
+ * @return The colour of tile in the small map in mode "Owner"
+ */
+static inline byte GetMinimapPixels(TileIndex tile)
+{
+	Owner o;
+
+	switch (GetTileType(tile)) {
+	case MP_INDUSTRY: o = OWNER_END;          break;
+	case MP_HOUSE:    o = OWNER_TOWN;         break;
+	default:          o = GetTileOwner(tile); break;
+		/* FIXME: For MP_ROAD there are multiple owners.
+		 * GetTileOwner returns the rail owner (level crossing) resp. the owner of ROADTYPE_ROAD (normal road),
+		 * even if there are no ROADTYPE_ROAD bits on the tile.
```

Comment:
I don't follow this comment. How can a level crossing be a level crossing if it doesn't have any roadtype bits?

Reply:
Please note that this code originally not made from me, so I feel sorry that I don't have much knowledge about it.
And I think this codes/comment is from here: https://github.com/OpenTTD/OpenTTD/blob/13cc8a0ceec90def39cbcb84135a0bf039793a6f/src/smallmap_gui.cpp#L573